server/libs/domain/src/search/search.service.ts

```diff
@@ -99,21 +99,18 @@ export class SearchService {
     this.assertEnabled();
 
     const query = dto.q || dto.query || '*';
-    const strategy = dto.clip ? SearchStrategy.CLIP : SearchStrategy.TEXT;
+    const strategy = dto.clip && MACHINE_LEARNING_ENABLED ? SearchStrategy.CLIP : SearchStrategy.TEXT;
```

Comment:
Does this need parens for the ternary to apply to both the first two conditions?

Reply:
I added parens at first because I preferred it for readability and the formatter removed them, so I assumed they're not needed for the conditional either.

Reply:
Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#table, `&&` takes precedence over ternaries.

Reply:
Ok perfect.